hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/StreamSync.java

```diff
@@ -584,10 +604,16 @@ private Pair<SchemaProvider, Pair<String, JavaRDD<HoodieRecord>>> fetchFromSourc
                 (SchemaProvider) new DelegatingSchemaProvider(props, hoodieSparkContext.jsc(), dataAndCheckpoint.getSchemaProvider(),
                     new SimpleSchemaProvider(hoodieSparkContext.jsc(), targetSchema, props)))
             .orElse(dataAndCheckpoint.getSchemaProvider());
+        if (rowBulkInsert) {
+          return new InputBatch(transformed, checkpointStr, schemaProvider);
```

Comment:
There is code handling empty commits and key gen following this code path, what happens if we return here ?

Reply:
we handle empty commits. wrt key gen, it will be handled in BaseDatasetBulkInsertCommitActionExecutor.